pulsar-package-manager/src/main/java/org/apache/pulsar/packages/manager/impl/PackageImpl.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pulsar.packages.manager.impl;
+
+import com.google.gson.Gson;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import org.apache.pulsar.common.naming.NamespaceName;
+import org.apache.pulsar.common.util.FutureUtil;
+import org.apache.pulsar.packages.manager.PackageMetadata;
+import org.apache.pulsar.packages.manager.PackageStorage;
+import org.apache.pulsar.packages.manager.Package;
+import org.apache.pulsar.packages.manager.exception.PackageAlreadyExistsException;
+import org.apache.pulsar.packages.manager.exception.PackageMetaAlreadyExistsException;
+import org.apache.pulsar.packages.manager.exception.PackageMetaNotFoundException;
+import org.apache.pulsar.packages.manager.exception.PackageNotFoundException;
+import org.apache.pulsar.packages.manager.naming.PackageName;
+import org.apache.pulsar.packages.manager.naming.PackageType;
+
+/**
+ * The implementation of package management.
+ */
+public class PackageImpl implements Package {
+
+    private final PackageStorage packageStorage;
+
+    public PackageImpl(PackageStorage packageStorage) {
+        this.packageStorage = packageStorage;
+    }
+
+    @Override
+    public CompletableFuture<PackageMetadata> getMeta(PackageName packageName) {
+        String metadataPath = getMetadataStoragePath(packageName);
+        return packageStorage.existAsync(metadataPath)
+            .thenCompose(exists -> exists
+                ? packageStorage.readAsync(metadataPath)
+                : FutureUtil.failedFuture(new PackageMetaNotFoundException("Package metadata does not existAsync")))
```

Comment:
1) What does "does not existAsync" mean here?
2) The error message doesn't contain any meaningful information. When a user receives such error message or see it in the log, he/she doesn't know what the error is. So please add meaningful information to let user know what package has the exception.

```suggestion
                : FutureUtil.failedFuture(new PackageMetaNotFoundException("The metadata of package " + packageName + " does not exist")))
```